RockPaperScissors.js

```diff
@@ -0,0 +1,95 @@
+var gamesWon = 0;
+var gamesPlayed = 0;
+var userName = " ";
+var gameState = "nameInput";
+
+var main = function (input) {
+  // Check Game Mode
+  if (gameState === "nameInput") {
+    userName = input;
+    gameState = "gameTime";
+    return "gm, " + userName + "! Welcome to the game!";
+  } else {
+    //Validate input is string
+    if (typeof input === "string") {
+      if (
+        input.toLowerCase() === "rock" ||
+        input.toLowerCase() === "paper" ||
+        input.toLowerCase() === "scissors"
```

Comment:
good that you have input validation here! maybe save the input.toLowercase() as a variable so you don't have to repeat it so many times